server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/aws/AwsS3ObjectStorage.java

```diff
@@ -90,7 +92,7 @@
         });
     }
 
-    private static void put(BlobId.Factory blobIdFactory, ContainerName containerName, AwsS3AuthConfiguration configuration, Blob blob, File file) {
+    private static void put(BlobId.Factory blobIdFactory, ContainerName containerName, AwsS3AuthConfiguration configuration, Blob blob, File file, int tried) {
```

Comment:
I'd rather prefer that you split error handling/retry mechanism and business logic.

Extract try/catch body and manage retry count at this level.